scripts/gha/test_simulator.py

```diff
@@ -745,7 +745,7 @@ def _run_android_test(testapp_dir, package_name, app_path, helper_project, max_a
     _uninstall_android_app(package_name)
 
     result = test_validation.validate_results(log, test_validation.CPP)
-    if result.complete and (FLAGS.test_type != "uitest" or result.fails == 0):
```

Comment:
What does this line fix?

Reply:
Retry if some tests failed or the log is incomplete.
Previously, only retry if the test log is incomplete.